erizo_controller/erizoJS/models/Publisher.js

```diff
@@ -129,24 +125,169 @@ class Source extends Node {
     return this.externalOutputs[url];
   }
 
-  muteStream(muteVideo, muteAudio) {
-    this.muteVideo = muteVideo;
-    this.muteAudio = muteAudio;
-    for (var subId in this.subscribers) {
-      var sub = this.getSubscriber(subId);
-      this.muteSubscriberStream(subId, sub.muteVideo, sub.muteAudio);
+  disableDefaultHandlers() {
+    const disabledHandlers = global.config.erizo.disabledHandlers;
+    if (!this.mediaStream) {
+      return;
+    }
+    for (const index in disabledHandlers) {
+      this.mediaStream.disableHandler(disabledHandlers[index]);
+    }
+  }
+
+  _onSchemeSlideShowModeChange(message, clientId) {
+    this.setSlideShow(message.enabled, clientId);
+  }
+
+  onSignalingMessage(msg) {
+    const connection = this.connection;
+    if (!connection) {
+      return;
+    }
+    if (msg.type === 'offer') {
+      const sdp = SemanticSdp.SDPInfo.processString(msg.sdp);
+      connection.setRemoteDescription(sdp);
+      this.disableDefaultHandlers();
+    } else if (msg.type === 'candidate') {
+      connection.addRemoteCandidate(msg.candidate);
+    } else if (msg.type === 'updatestream') {
+      if (msg.sdp) {
+        const sdp = SemanticSdp.SDPInfo.processString(msg.sdp);
+        connection.setRemoteDescription(sdp);
+      }
+      if (msg.config) {
+        if (msg.config.minVideoBW) {
+          log.debug('message: updating minVideoBW for publisher, ' +
+                    'id: ' + this.streamId + ', ' +
+                    'minVideoBW: ' + msg.config.minVideoBW);
+          this.minVideoBW = msg.config.minVideoBW;
+          for (const clientId in this.subscribers) {
+            const subscriber = this.getSubscriber(clientId);
+            subscriber.minVideoBW = msg.config.minVideoBW * 1000; // bps
+            subscriber.lowerThres = Math.floor(subscriber.minVideoBW * (1 - 0.2));
+            subscriber.upperThres = Math.ceil(subscriber.minVideoBW * (1 + 0.1));
+          }
+        }
+        if (msg.config.muteStream !== undefined) {
+          this.muteStream(msg.config.muteStream);
+        }
+      }
+    } else if (msg.type === 'control') {
+      this.processControlMessage(undefined, msg.action);
+    }
+  }
+
+  processControlMessage(clientId, action) {
+    const publisherSide = clientId === undefined || action.publisherSide;
+    switch(action.name) {
+      case 'controlhandlers':
+        if (action.enable) {
+          this.enableHandlers(publisherSide ? undefined : clientId, action.handlers);
+        } else {
+          this.disableHandlers(publisherSide ? undefined : clientId, action.handlers);
+        }
+        break;
+    }
+  }
+
+  generateVideoKeyFrame() {
```

Comment:
Another nitpick. Since we're not actually generating a keyframe but requesting one via PLI. Can we name this `requestVideoKeyFrame`?

Reply:
you got it